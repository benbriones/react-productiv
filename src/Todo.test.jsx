import {describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

describe("todo", function () {
  it("renders without crashing", function () {
    render(<Todo id={1} title="cook" description="dinner" priority={1}/>);
  });

  it("matches snapshot", function () {
    const { container } = render(<Todo id={1} title="cook" description="dinner" priority={1}/>)
    expect(container).toMatchSnapshot();
  });

  it ("rendered contains expected", function () {
    const { container } = render(<Todo id={1} title="cook" description="dinner" priority={1}/>)
    const toDo = container.querySelector(".Todo")
    expect(toDo).toContainHTML("cook");
  });
});

