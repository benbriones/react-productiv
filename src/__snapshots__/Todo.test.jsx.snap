// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`todo > matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        cook
      </b>
       
      <small>
        (Priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        dinner
      </small>
    </div>
  </div>
</div>
`;
